data_node/controllers/outer/upload_controller.go

```diff
@@ -396,9 +396,16 @@ func (server *Server) handleAppendUpload(w http.ResponseWriter, r *http.Request)
 	}
 
 	if fileInfo.Offset == fileInfo.Size {
+		if fileInfo.Type == datanode.VideoFileType {
+			if !server.isReplica(fileInfo) {
+				go startJob(fileInfo)
```

Comment:
What's your rationale for using a different thread? We already run the command in the background so it's non-blocking.